sympy/diffgeom/diffgeom.py

```diff
@@ -43,14 +44,23 @@ class Manifold(Atom):
     """
 
     def __new__(cls, name, dim):
-        obj = super().__new__(cls)
-        obj.name = name
-        obj.dim = dim
+        if not isinstance(name, Str):
+            name = Str(name)
+        dim = _sympify(dim)
+        obj = super().__new__(cls, name, dim)
         obj.patches = []
         # The patches list is necessary if a Patch instance needs to enumerate
         # other Patch instance on the same manifold.
         return obj
 
+    @property
+    def name(self):
+        return self.args[0]
+
+    @property
+    def dim(self):
+        return self.args[1]
+
     def _hashable_content(self):
```

Comment:
You won't have to redefine hashable_content

Reply:
Oops. I forgot to remove that.